src/types/user_operation.rs

```diff
@@ -17,3 +22,161 @@ pub struct UserOperation {
+                nonce: U256::zero(),
+                init_code: Bytes::default(),
+                call_data: Bytes::default(),
+                call_gas_limit: U256::from(200000),
+                verification_gas_limit: U256::from(100000),
+                pre_verification_gas: U256::from(21000),
+                max_fee_per_gas: U256::from(3000000000 as u64),
+                max_priority_fee_per_gas: U256::from(1000000000),
+                paymaster_and_data: Bytes::default(),
+                signature: Bytes::from_str("0x7cb39607585dee8e297d0d7a669ad8c5e43975220b6773c10a138deadbc8ec864981de4b9b3c735288a217115fb33f8326a61ddabc60a534e3b5536515c70f931c").unwrap(),
+            },
+        ];
+        assert_eq!(user_operations[0].pack(), String::from("0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001600000000000000000000000000000000000000000000000000000000000000180000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000186a000000000000000000000000000000000000000000000000000000000000052080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003b9aca0000000000000000000000000000000000000000000000000000000000000001a000000000000000000000000000000000000000000000000000000000000001c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000").into_bytes());
+        assert_eq!(user_operations[1].pack(), String::from("0x000000000000000000000000663f3ad617193148711d28f5334ee4ed070166020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000016000000000000000000000000000000000000000000000000000000000000001800000000000000000000000000000000000000000000000000000000000030d4000000000000000000000000000000000000000000000000000000000000186a0000000000000000000000000000000000000000000000000000000000000520800000000000000000000000000000000000000000000000000000000b2d05e00000000000000000000000000000000000000000000000000000000003b9aca0000000000000000000000000000000000000000000000000000000000000001a000000000000000000000000000000000000000000000000000000000000001c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000417cb39607585dee8e297d0d7a669ad8c5e43975220b6773c10a138deadbc8ec864981de4b9b3c735288a217115fb33f8326a61ddabc60a534e3b5536515c70f931c00000000000000000000000000000000000000000000000000000000000000").into_bytes());
```

Comment:
I am just curious here. Where are these test cases (and below test cases) coming from? I was looking at the typescript aa repo but didn't find.

Reply:
First test case: https://github.com/eth-infinitism/account-abstraction/blob/develop/test/UserOp.ts#L113
Second test case is taken from this script: https://github.com/eth-infinitism/account-abstraction/blob/develop/test/simple-wallet.test.ts#L67

Reply:
Thanks~! I figure that out now~! Good work on this